server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -1330,6 +1339,20 @@ protected void doClose() {
         clusterService.removeApplier(this);
     }
 
+    /**
+     * Assert that no in-memory state for any running snapshot operation exists in this instance.
+     */
+    public boolean assertAllListenersResolved() {
```

Comment:
Just to be sure, do we already assert somewhere else that there are not listeners leaking after any cluster state update?

Reply:
Replying to myself: we do, in `SnapshotsService.assertConsistentWithClusterState()`

Reply:
Yea sadly that assertion (and the way it works in general) can only make sure that we are not failing to resolve a listener during an action (i.e. updating the state because of some other state change or shard update message). It unfortunately couldn't detect that the final state of things is what we want.